partner_phone_search/models/res_partner.py

```diff
@@ -0,0 +1,19 @@
+# Copyright 2018 - TODAY Serpent Consulting Services Pvt. Ltd.
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, models
+
+
+class ResPartner(models.Model):
+    _inherit = 'res.partner'
+
+    @api.model
+    def name_search(self, name, args=None, operator='ilike', limit=100):
+        if name:
+            domain = ['|', '|', ('phone', operator, name),
```

Comment:
I remember to have problems implementing something similar in 

https://github.com/OCA/l10n-spain/blob/dc1d2add9990a27f711901149d2a9fbfb9dda24c/l10n_es_partner/models/res_partner.py#L39

And I finally gave up giving precedence to my query.